tests/unit/s2n_extensions_server_key_share_select_test.c

```diff
@@ -383,8 +383,7 @@ int main()
             EXPECT_SUCCESS(s2n_connection_free(server_conn));
         };
 
-        /* When client sent valid keyshares
-         * only for ECC, server should choose curves[0] and not send HRR. */
+        /* When client sent KeyShares only for ECC but also supports PQ, server should choose PQ and send HRR. */
```

Comment:
It makes sense to me to retry if the client sent an invalid PQ keyshare but supports valid PQ, but not to retry if the client didn't send any PQ. Are you sure about this behavior? Wouldn't no PQ share indicate that the client doesn't really want to do PQ?

It also seems like that would meet the usecase of upgrading from one PQ algorithm to another, without triggering retries for clients that might not care about PQ.

Reply:
> Are you sure about this behavior? Wouldn't no PQ share indicate that the client doesn't really want to do PQ?
No, it wouldn't. There may be clients who do not want to waste unnecessary bandwidth sending PQ KeyShares if they've previously tested and confirmed the endpoints they are connecting to don't **yet** support PQ, but might support PQ at some point in the future. These clients can decide to indicate that they support PQ in there SupportedGroups extension (2 bytes added to the handshake), without including PQ KeyShare (800-1,568 bytes added) in their KeyShare extension, and only include an ECDHE KeyShare. This behavior is allowed by the TLS 1.3 RFC, and if a client adopt this behavior, those clients would never be able to gracefully upgrade to PQ when server-side PQ deployments occur.